executor/aggfuncs/func_percentile.go

```diff
@@ -0,0 +1,325 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aggfuncs
+
+import (
+	"math"
+	"sort"
+	"unsafe"
+
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pingcap/tidb/util/selection"
+)
+
+const (
+	// DefPartialResult4PercentileIntSize is the size of partialResult4PercentileInt
+	DefPartialResult4PercentileIntSize = int64(unsafe.Sizeof(partialResult4PercentileInt{}))
+	// DefPartialResult4PercentileRealSize is the size of partialResult4PercentileReal
+	DefPartialResult4PercentileRealSize = int64(unsafe.Sizeof(partialResult4PercentileReal{}))
+	// DefPartialResult4PercentileDecimalSize is the size of partialResult4PercentileDecimal
+	DefPartialResult4PercentileDecimalSize = int64(unsafe.Sizeof(partialResult4PercentileDecimal{}))
+	// DefPartialResult4PercentileTimeSize is the size of partialResult4PercentileTime
+	DefPartialResult4PercentileTimeSize = int64(unsafe.Sizeof(partialResult4PercentileTime{}))
+)
+
+func percentile(data sort.Interface, percent int) int {
+	// Ordinal rank k = Ceil(P / 100 * N)
+	k := int(math.Ceil(float64(data.Len()*percent) / 100.0))
```

Comment:
1. In common case, k would never overflow MaxInt32. If it does exceed MaxInt32, our algorithm will break since golang doesn't support a slice with size larger than MaxInt32.
2. I'll fix it.